crates/astria-sequencer-relayer/src/relayer.rs

```diff
@@ -97,62 +92,85 @@ impl Relayer {
         self.state_tx.subscribe()
     }
 
-    async fn get_and_submit_latest_block(&self) -> eyre::Result<State> {
-        let mut new_state = (*self.state_tx.borrow()).clone();
-        let resp = self.sequencer_client.get_latest_block().await?;
-
-        let height = resp.block.header.height.value();
-        if height <= *new_state.current_sequencer_height.get_or_insert(height) {
-            return Ok(new_state);
-        }
-
-        info!(height = ?height, tx_count = resp.block.data.len(), "got block from sequencer");
-        new_state.current_sequencer_height.replace(height);
-
-        if resp.block.header.proposer_address.as_ref() != self.validator.address.as_ref() {
-            let proposer_address = resp.block.header.proposer_address;
-            info!(
-                %proposer_address,
-                validator_address = %self.validator.address,
-                "ignoring block: proposer address is not ours",
+    fn convert_block_response_to_sequencer_block_data(
+        &self,
+        res: BlockResponse,
+        current_height: Option<u64>,
+    ) -> eyre::Result<SequencerBlockData> {
+        if let Some(current_height) = current_height {
+            ensure!(
+                res.block.header.height.value() > current_height,
+                "sequencer block response had height below current height tracked in relayer"
             );
-            return Ok(new_state);
         }
+        ensure!(
+            res.block.header.proposer_address == self.validator.address,
+            "proposer recorded in sequencer block does not match internal validator"
+        );
```

Comment:
ACK, this will be very noisy. I was reaching for an early return. I will change the logic to log this at the debug level and not overuse error handling.